### PR TITLE
add kubectl fish shell completion

### PR DESCRIPTION
**Pull request by WLun001**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test

/kind feature
> /kind flake

**What this PR does / why we need it**:
add fish shell completion, `kubectl completion fish`

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Add fish shell completion to kubectl
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by WLun001** [comment]
I signed it

**Comment by marckhouzam** [comment]
This is great!
I believe @brianpursley had looked at this, so he might be interested.

**Comment by eddiezane** [comment]
/ok-to-test

@brianpursley @knight42 not sure where this stands with what y'all were cooking up.

**Comment by WLun001** [comment]
Not sure how to fix the failing test on `kubelet` as I did not modified it. Please advise

**Comment by eddiezane** [comment]
@WLun001 can you try rebasing off master please?

**Comment by eddiezane** [comment]
@WLun001 it looks like master was merged into your branch. Might be easiest to reset your master branch and start with a fresh branch for this PR.

https://github.com/kubernetes/kubernetes/commit/c89d070bdea18992a135da8cbeb2d57211d9f101

**Comment by liggitt** [comment]
> @WLun001 it looks like master was merged into your branch. Might be easiest to reset your master branch and start with a fresh branch for this PR.

no need to restart, the force-push fixed it

**Comment by WLun001** [comment]
@eddiezane @liggitt I have rebased, but the test still failed. Should I start with a new PR and branch?

**Comment by marckhouzam** [comment]
This looks ready I think. It would be a nice addition that's been requested by users for quite a while.

**Comment by WLun001** [comment]
@eddiezane @liggitt any ETA for this to merge?

**Comment by eddiezane** [comment]
This shouldn't interfere with any of the ongoing completion work but I'll ping @brianpursley to confirm.

/assign @soltysh

**Comment by brianpursley** [comment]
lgtm, but keep in mind this will only provide static completions, not dynamic.  So it will complete command names and flags, but will not show a list of resources for example.

**Review by brianpursley — APPROVED** [review]

**Comment by marckhouzam** [comment]
> lgtm, but keep in mind this will only provide static completions, not dynamic. So it will complete command names and flags, but will not show a list of resources for example.

@brianpursley @eddiezane I've removed all bash scripting in #96087.  It should allow Fish completion to fully benefit from kubectl's custom completions.

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
Looks like a good change that can be merged soon?

**Comment by marckhouzam** [comment]
Should this be merged as is?  Static completions would still be nice while waiting for #96087 to get some attention.

**Comment by marckhouzam** [comment]
With #96087 having been merged, this PR would benefit from a rebase which will provide complete Fish completion.

**Comment by brianpursley** [comment]
Fixes https://github.com/kubernetes/kubectl/issues/576

File: staging/src/k8s.io/kubectl/pkg/cmd/completion/completion.go [hunk 1]
```diff
@@ -82,13 +82,20 @@ var (
 		# Load the kubectl completion code for zsh[1] into the current shell
 		    source <(kubectl completion zsh)
 		# Set the kubectl completion code for zsh[1] to autoload on startup
-		    kubectl completion zsh > "${fpath[1]}/_kubectl"`))
+		    kubectl completion zsh > "${fpath[1]}/_kubectl"
+
+
+		# Load the kubectl completion code for fish[2] into the current shell
+		 	kubectl completion fish | source
```
Comment (knight42) [inline staging/src/k8s.io/kubectl/pkg/cmd/completion/completion.go:89]: nit: Could you make the indentation depth consistent with the above text?

**Review by knight42 — APPROVED** [review]
/lgtm

Thanks!

**Review by knight42 — APPROVED** [review]
/lgtm

**Review by eddiezane — APPROVED** [review]

**Pull request merged**